v3/src/components/graph/components/scatterdots.tsx

```diff
@@ -19,12 +19,12 @@ export const ScatterDots = function ScatterDots(props: PlotProps) {
     instanceId = useInstanceIdContext(),
     dataConfiguration = useDataConfigurationContext(),
     dataset = useDataSetContext(),
+    primaryAttrIDRef = useRef<string>(''),
+    secondaryAttrIDsRef = useRef<string[]>([]),
```

Comment:
You do need the `<string[]>` in the second case because the type can't be inferred from the default value.